src/lib/core/exception.cc

```diff
@@ -425,3 +451,25 @@ BuildFileFormatError(const char *file, unsigned line, const char *format, ...)
 	va_end(ap);
 	return e;
 }
+
+struct error *
+BuildEncodeError(const char *file, unsigned line, const char *format,
+		 const char *details)
+{
+	EncodeError *e = new EncodeError(file, line, "");
+	error_format_msg(e, "%s encode error: %s", format, details);
```

Comment:
I think you could write:

```C
	EncodeError *e = new EncodeError(file, line, "%s encode error: %s", format details);
```

Same comment for `DecodeError`.

Reply:
Done.